server/src/graphql/resolvers/userResolvers.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable indent */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { GraphQLError } from 'graphql';
+import { Response } from 'express';
+
+import { MyContext, UserResolverInterface } from '../../types';
+
+import { RESPONSE_MESSAGE } from '../../constants';
+
+import { userEntity } from '../../entities';
+import HttpStatusCodes from '../../constants/HttpStatusCodes';
+
+export const userResolver: UserResolverInterface = {
+  userAvatar: async (_: Request, __: Response, { userId }: MyContext) => {
+    if (!userId) {
+      throw new GraphQLError(RESPONSE_MESSAGE.USER.FAIL.NOT_AUTH, {
+        extensions: { code: HttpStatusCodes.UNAUTHORIZED },
+      });
+    }
+
+    const user = await userEntity.findUserById(userId);
+
+    if (!user) {
+      throw new GraphQLError(RESPONSE_MESSAGE.USER.FAIL.NOT_EXIST, {
+        extensions: { code: HttpStatusCodes.UNAUTHORIZED },
+      });
+    }
+
+    return { avatar: user.avatar };
+  },
+
+  userData: async (_: Request, __: Response, { userId }: MyContext) => {
+    if (!userId) {
```

Comment:
Why you checking it in each method? You can check it in schema, as mandatory field and then `gq` by itself will throw an error if `userId` is not passed to.

Reply:
`userId: String!`

Reply:
Understood, how do you do. Then you can create composition, in which you will check such things from `context` - create and wrap your resolver method into it.